pkg/healthchecker/types/types.go

```diff
@@ -38,12 +39,51 @@ const (
 	ContainerdService  = "containerd"
 	KubeProxyComponent = "kube-proxy"
 
-	KubeletHealthCheckEndpoint   = "http://127.0.0.1:10248/healthz"
-	KubeProxyHealthCheckEndpoint = "http://127.0.0.1:10256/healthz"
-
 	LogPatternFlagSeparator = ":"
+
+	nodeEnvKey       = "HOST_IP"
+	kubeletPortKey   = "KUBELET_PORT"
+	kubeProxyPortKey = "KUBEPROXY_PORT"
+
+	defaultHostIP        = "127.0.0.1"
+	defaultKubeletPort   = "10248"
+	defaultKubeproxyPort = "10256"
+)
+
+var (
```

Comment:
Since these values are now public variables, they can be overwritten by other means. The only way I can think of preventing this is to have private strings and public functions that return those private values. Eg.
```go
var kubeletHealthCheckEndpoint string
// Assign the value via setKubeEndpoints

func KubeletHealthCheckEndpoint() string { return kubeletHealthCheckEndpoint }
```